### PR TITLE
Setup Cypress e2e tests

### PR DESCRIPTION
**Pull request by fatonramadani**
Initial setup of Cypress.
This PR adds:
- 2 simple authentication tests: Login and Redirect

Tests can be run in two ways:
- `npm run cypress:run` : Headlessly in the terminal
- `npm run cypress:open `: Opens up the Cypress GUI

An upcomming PR will run Cypress tests with GitHub Actions as part of a CI/CD pipeline.

### CODE REVIEW

File: frontend/cypress/integration/authentication.spec.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+describe('Authentication', () => {
+	it('can login using email and password', () => {
+		cy.visit(`${Cypress.env('baseUrl')}/user/login`)
+
+		cy.get('#showPassword').click()
+		cy.get('#email').type('admin@windmill.dev')
+		cy.get('#password').type('changeme')
+		cy.get('.flex > .default-button').click()
+
+		cy.contains('Select a workspace')
```
Comment (fatonramadani) [inline frontend/cypress/integration/authentication.spec.ts:11]: @rubenfiszel we should probably use i18n to avoid these kinds of string duplication.

Comment (rubenfiszel) [reply frontend/cypress/integration/authentication.spec.ts:11]: I thought i18n was mostly about internationalization. What do you have in mind ?

File: frontend/cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add('login', (email, password) => { ... })
```
Comment (rubenfiszel) [inline frontend/cypress/support/commands.js]: Might it make sense to have authentication as a command and change your test to an empty one just using that command for now ?

File: frontend/cypress.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+	"pluginsFile": false,
+	"env": {
+		"baseUrl": "http://localhost"
```
Comment (rubenfiszel) [inline frontend/cypress.json]: Using the docker, you might want to it port 8000 directly

File: frontend/cypress/support/commands.js (hunk 2 above)
Comment (fatonramadani) [reply frontend/cypress/support/commands.js]: It's done.

**Pull request merged**